src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/finance/PrisonerBalanceReconciliationService.kt

```diff
@@ -161,12 +165,15 @@ class PrisonerBalanceReconciliationService(
 
       is AccountFields -> {
         nomisObj as AccountFields
+        if (dpsObj.prisonId.compareTo(nomisObj.prisonId) != 0) {
+          differences.add(Difference("$parentProperty.prisonId", dpsObj.prisonId, nomisObj.prisonId))
+        }
         if (dpsObj.balance.compareTo(nomisObj.balance) != 0) {
           differences.add(Difference("$parentProperty.balance", dpsObj.balance, nomisObj.balance))
         }
         if (
           // DPS holdBalance is non-null
-          nomisObj.holdBalance == null ||
+          (nomisObj.holdBalance == null && dpsObj.holdBalance!! != BigDecimal.ZERO) ||
```

Comment:
During migratoin, null holdBalance in Nomis is converted to zero when passed to DPS (so ensure this does not cause mismatch)